### PR TITLE
Update context menus in Pages list Action

### LINKED ISSUE #19356: UI Modernization Pages: Update context menus
```[tasklist]
### Tasks
- [x] Apply new grouping and ordering for the context menu on each tab
- [x] Adjust copy and icons
- [x] Apply existing visibility rules
```

### PR DESCRIPTION
**Pull request by AjeshRPai**
## Closes
#19356 

## Description
This PR
1.  ↑ Updates:  the Context menu of the pages list actions with divider and new icons 
2. ↑ Updates: the order of the pages list actions according to the design 

## 📷  Screenshots
|  Published by Admin | Published by Contributor  | Scheduled | Draft  | Trash   |
|---|---|---|---|---|
|  Screenshot 2023-11-02 at 1 38 38 PM |Screenshot 2023-11-02 at 1 38 50 PM |Screenshot 2023-11-02 at 1 40 09 PM | Screenshot 2023-11-02 at 1 39 02 PM|Screenshot 2023-11-02 at 1 39 12 PM |

📓 There is no changes to functionalities of the pages list item actions  

### Prereqs
Have a few accounts with editor, contributor, and admin capabilities

### Test
1. Create a fews Pages as a contributor - drafts, scheduled, and non-scheduled (submit these)
2. Create a few Pages as editor/admin
3. Install the JP app
4. Launch and sign in
5. Navigate to My Site > Pages
6. Tap on the "more" icon on a pages row in the Published tab
7. ✅ Verify the new icon groupings are shown, the icons and copy are correct
8. Repeat the last two steps for each tab

## ⚠️  Merge Instructions
1. Wait for the https://github.com/wordpress-mobile/WordPress-Android/pull/19498 to merge
2. Make sure that the base branch changes to `feature/ui-modernization-posts-and-pages` or to `trunk` - depending on the merge of feature branch PR 
3. Remove Do not merge label 
4. Merge as normal

## Regression Notes
1. Potential unintended areas of impact
Icons are not shown as intended and not grouped as intended

5. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

6. What automated tests I added (or what prevented me from doing so)
Manual testing 

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/pages/PageItem.kt [hunk 1]
```diff
@@ -213,6 +213,15 @@ sealed class PageItem(open val type: Type) {
         VIRTUAL_HOMEPAGE(6),
     }
 
+    enum class PositionGroup(val position:Int){
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/pages/PageItem.kt]: Nice idea to use an enum. ❤️

**Review by zwarm — CHANGES_REQUESTED** [review]
@AjeshRPai - Like the refactors for the emptyDrawable - going to apply to the posts list. 

I'm going to ask for a change to the icon names, so all "gb" icons use the same pattern "gb_ic_{name of gb icon}"? This way in the future we know which are GB icons and which aren't.

Thanks

**Comment by AjeshRPai** [comment]
Hey @zwarm 

>I'm going to ask for a change to the icon names, so all "gb" icons use the same pattern "gb_ic_{name of gb icon}"? This way in the future we know which are GB icons and which aren't.

💯  Agree 👍🏼 . I have renamed the icons to use gb with the commit [70ef253](https://github.com/wordpress-mobile/WordPress-Android/pull/19513/commits/70ef253c296bdf8f13acafe7cbfbb47851cab238), I am not renaming the existing icons. ie blaze and posts_set_as_parent which are used in the pages list actions.

**Review by zwarm — APPROVED** [review]
@AjeshRPai - Thanks for making those changes. I have approved. @osullivanchris - I am going to merge this PR into our feature branch so we push everything to trunk. Any design issues will be taken up in a new PR.

**Pull request merged**